Use WPE_RENDERER on PlayStation

donny-dont opened the pull request:
#### 5157a857f440e04827ccbd98112f219b99caef61
<pre>
Use WPE_RENDERER on PlayStation
<a href="https://bugs.webkit.org/show_bug.cgi?id=248480">https://bugs.webkit.org/show_bug.cgi?id=248480</a>

Reviewed by Adrian Perez de Castro.

When `USE_WPE_BACKEND_PLAYSTATION` turn on `USE_WPE_RENDERER`.

* Source/WebKit/UIProcess/playstation/PageClientImpl.cpp
* Source/WebKit/UIProcess/playstation/PageClientImpl.h
* Source/WebKit/UIProcess/playstation/WebProcessPoolPlayStation.cpp
* Source/WebKit/WebProcess/EntryPoint/playstation/WebProcessMain.cpp
* Source/WebKit/WebProcess/playstation/WebProcessPlayStation.cpp
* Source/cmake/OptionsPlayStation.cmake

Canonical link: <a href="https://commits.webkit.org/262586@main">https://commits.webkit.org/262586@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/67f79240b78a7db14b0ffdd11577ded3c7cc33fc

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/1958 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/26/builds/1989 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/14/builds/2051 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/2885 "Built successfully") | [✅ 🛠 wincairo](https://ews-build.webkit.org/#/builders/32/builds/2046 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/23/builds/2066 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/16/builds/2035 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/1805 "Passed tests") | 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/1977 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/15/builds/1768 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/1753 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/2731 "Built successfully") | 
| | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/1758 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/10/builds/1735 "Passed tests") | [❌ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/1651 "27 flakes 150 failures") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/1647 "Built successfully and passed tests") | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/7/builds/1802 "Built successfully") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/36/builds/1782 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/2908 "Passed tests") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/1877 "Built successfully and passed tests") | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/30/builds/1814 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/17/builds/1613 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/2018 "Built successfully") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/4/builds/1741 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/1746 "Passed tests") | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/459 "Passed tests") | 
| [  ~~🛠 🧪 merge~~](https://ews-build.webkit.org/#/builders/19/builds/487 "The change is no longer eligible for processing.") | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/31/builds/1905 "Built successfully") | | [✅ 🛠 jsc-mips](https://ews-build.webkit.org/#/builders/24/builds/2063 "Built successfully") | 
| [✅ 🛠 🧪 unsafe-merge](https://ews-build.webkit.org/#/builders/22/builds/214 "Built successfully and passed tests") | | | [✅ 🧪 jsc-mips-tests](https://ews-build.webkit.org/#/builders/3/builds/574 "Passed tests") | 
<!--EWS-Status-Bubble-End-->

aperezdc commented on Source/WebKit/UIProcess/playstation/PageClientImpl.cpp:
```diff
@@ -344,4 +348,11 @@ void PageClientImpl::requestDOMPasteAccess(WebCore::DOMPasteAccessCategory, cons
+#if USE(WPE_RENDERER)
+UnixFileDescriptor PageClientImpl::hostFileDescriptor()
+{
+    return UnixFileDescriptor { 1, UnixFileDescriptor::Adopt };
```
As discussed by chat, this is a stub implementation, so you may want to add a `TODO` comment here as a reminder&mdash;up to you.

aperezdc reviewed (commented):
r=me

donny-dont committed:
Use WPE_RENDERER on PlayStation
https://bugs.webkit.org/show_bug.cgi?id=248480

Reviewed by Adrian Perez de Castro.

When `USE_WPE_BACKEND_PLAYSTATION` turn on `USE_WPE_RENDERER`.

* Source/WebKit/UIProcess/playstation/PageClientImpl.cpp
* Source/WebKit/UIProcess/playstation/PageClientImpl.h
* Source/WebKit/UIProcess/playstation/WebProcessPoolPlayStation.cpp
* Source/WebKit/WebProcess/EntryPoint/playstation/WebProcessMain.cpp
* Source/WebKit/WebProcess/playstation/WebProcessPlayStation.cpp
* Source/cmake/OptionsPlayStation.cmake

Canonical link: https://commits.webkit.org/262586@main
